server/api/views/shift.py

```diff
@@ -49,33 +49,33 @@ def put(self,request):
             creator_member = WorkspaceMember.objects.get(user=request.user, workspace=workspace)
             creator_permissions = MemberPermissions.objects.get(member=creator_member, MANAGE_SCHEDULES=True)
         except WorkspaceMember.DoesNotExist:
-            response["error"]["message"] = "You are not a member of this workspace"
+            response["error"]["message"] = "You are not a member of this workspace."
             return Response(response, status=status.HTTP_403_FORBIDDEN)
         except MemberPermissions.DoesNotExist:
-            response["error"]["message"] = "You do not have permissions to manage schedules in this workspace"
+            response["error"]["message"] = "You do not have permissions to manage schedules in this workspace."
             return Response(response, status=status.HTTP_403_FORBIDDEN)
         
         # Verify role exists and is part of workspace
         try:
             role = WorkspaceRole.objects.get(pk=request.data["role_id"], workspace=workspace)
         except WorkspaceRole.DoesNotExist:
-            response["error"]["message"] = "Workspace role does not exist or is not part of workspace"
+            response["error"]["message"] = "Workspace role does not exist or is not part of workspace."
             return Response(response, status=status.HTTP_404_NOT_FOUND)
         
         # Verify member is valid if present
         if "member_id" in request.data:
             try:
                 member = WorkspaceMember.objects.get(pk=request.data["member_id"], workspace=workspace)
             except WorkspaceMember.DoesNotExist:
-                response["error"]["message"] = "Member does not exist or is not part of workspace"
+                response["error"]["message"] = "Member does not exist or is not part of workspace."
                 return Response(response, status=status.HTTP_404_NOT_FOUND)
         
         # verify dates are valid
         start_time = serializer.validated_data['start_time']
         end_time = serializer.validated_data['end_time']
 
         if start_time > end_time:
-            response["error"]["message"] = "Start time cannot be after end time"
+            response["error"]["message"] = "Start time cannot be after end time."
             return Response(response, status=status.HTTP_400_BAD_REQUEST) # idk if this is the right status code but wtvr
         
         # could check if start time is before current time if we want to prevent creating shifts in the past, but i think we should allow that since a workplace might want to do that for recordkeeping or smth
```

Comment:
Yeah that sounds like a good idea to allow that to happen. Another use case could be if an employee wasn't scheduled, but was asked verbally to come in and work (if it turned out to be busy, needed extra help with something, etc.) and the workplace wanted to record that shift after the fact.